src/javascript/components/SitemapPanelApp.jsx

```diff
@@ -0,0 +1,277 @@
+
+import styles from './SitemapPanel.scss';
+
+import {Card, Snackbar} from '@material-ui/core';
+
+import {compose} from '~/utils';
+import {withApollo} from 'react-apollo';
+import {withTranslation} from 'react-i18next';
+
+import * as gqlMutations from './gqlMutations';
+import * as gqlQueries from './gqlQueries';
+import * as gqlUtilities from '../utils/gqlUtilities';
+
+import {DialogComponent} from './Dialog';
+import {useFormik} from 'formik';
+
+const SitemapPanelApp = ({client, dxContext, t}) => {
+    const [sitemapMixinEnabled, setSitemapMixinEnabled] = useState(false);
+    const [sitemapIndexURL, setSitemapIndexURL] = useState(null);
+    const [sitemapCacheDuration, setSitemapCacheDuration] = useState(null);
+
+    const [dialogIsOpen, setDialogIsOpen] = useState(false);
+    const [dialogInfo, setDialogInfo] = useState(null);
+
+    const [snackbarIsOpen, setSnackbarIsOpen] = useState(false);
+
+    const dropdownData = [{
+        label: '4 hours',
+        value: '4h'
+    }, {
+        label: '8 hours',
+        value: '8h'
+    }, {
+        label: '24 hours',
+        value: '24h'
+    }, {
+        label: '48 hours',
+        value: '48h'
+    }];
+
+    const convertCacheDurationToLabel = durationValue => {
+        switch (durationValue) {
+            case dropdownData[1].value:
+                return dropdownData[1].label;
+            case dropdownData[2].value:
+                return dropdownData[2].label;
+            case dropdownData[3].value:
+                return dropdownData[3].label;
+            default:
+                return dropdownData[0].label;
+        }
+    };
+
+    const formik = useFormik({
+        initialValues: {
+            sitemapIndexURL: sitemapIndexURL,
+            sitemapCacheDuration: ((sitemapCacheDuration) ? sitemapCacheDuration : dropdownData[0].value)
+        },
+        enableReinitialize: true,
+        // eslint-disable-next-line no-unused-vars
+        onSubmit: values => {
+            if (!sitemapMixinEnabled) {
+                gqlUtilities.gqlMutate(client, gqlMutations.AddMixin, {
+                    pathOrId: `/sites/${dxContext.siteKey}`,
+                    mixins: ['jseomix:sitemap']
+                });
+                setSitemapMixinEnabled(prevState => !prevState);
+                setSnackbarIsOpen(true);
+            }
+
+            gqlUtilities.gqlMutate(client, gqlMutations.mutateProperty, {
+                pathOrId: `/sites/${dxContext.siteKey}`,
+                propertyName: 'sitemapIndexURL',
+                propertyValue: formik.values.sitemapIndexURL
+            });
+            gqlUtilities.gqlMutate(client, gqlMutations.mutateProperty, {
+                pathOrId: `/sites/${dxContext.siteKey}`,
+                propertyName: 'sitemapCacheDuration',
+                propertyValue: formik.values.sitemapCacheDuration
+            });
+        }
+    });
+
+    useEffect(() => {
```

Comment:
I guess if we use useQuery and pass siteKey as param it doesn't rerun query when siteKey changes?

Reply:
To be honest I haven't tried to use useQuery at this case. (Due to I re-factored from class component)
But being said ye there is a dependency on the siteKey we need to get the data again and update the page.
As we do have an option to switch the siteKey

Reply:
Checked that it doesn't work very well with the useEffect on the data being returned